datadog/resource_datadog_synthetics_test_.go

```diff
@@ -323,39 +328,39 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 	d.Set("type", syntheticsTest.GetType())
 
 	actualRequest := syntheticsTest.GetConfig().Request
-	localRequest := make(map[string]string)
+	localRequest := schema.ResourceData{}
 	if actualRequest.HasBody() {
-		localRequest["body"] = actualRequest.GetBody()
+		localRequest.Set("body", actualRequest.GetBody())
 	}
 	if actualRequest.HasMethod() {
-		localRequest["method"] = actualRequest.GetMethod()
+		localRequest.Set("method", actualRequest.GetMethod())
 	}
 	if actualRequest.HasTimeout() {
-		localRequest["timeout"] = convertToString(actualRequest.GetTimeout())
+		localRequest.Set("timeout", actualRequest.GetTimeout())
 	}
 	if actualRequest.HasUrl() {
-		localRequest["url"] = actualRequest.GetUrl()
+		localRequest.Set("url", actualRequest.GetUrl())
 	}
-	d.Set("request", localRequest)
+	d.Set("request", &localRequest)
 	d.Set("request_headers", actualRequest.Headers)
 
 	actualAssertions := syntheticsTest.GetConfig().Assertions
-	localAssertions := []map[string]string{}
+	localAssertions := []*schema.ResourceData{}
```

Comment:
Same, why considering an assertion as a resource?

Reply:
Intention was to avoid converting to string, but this does not pass acceptance tests, so reverting.